File: CPP/Ram.cpp
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"

using namespace std;

// Kelas Ram yang mewarisi dari kelas Komponen
class Ram : public Komponen
{
private:
    int kapasitasGB;
    string ddr;
public:
    Ram()
    {


    }

    // Konstruktor dengan parameter untuk menginisialisasi kapasitasGB, ddr, merk, dan nama
    Ram(int kapasitasGB, string ddr, string merk, string nama) : Komponen(merk, nama)
    {
        this->kapasitasGB = kapasitasGB;
        this->ddr = ddr;
    }

    //Setter
    void setKapasitasGB(int kapasitasGB)
    {
        this->kapasitasGB = kapasitasGB;
    }


    void setDdr(float ddr)
    {
        this->ddr = ddr;
    }


    //Getter
    int getKapasitasGB()
    {
        return this->kapasitasGB;
    }


    string getDdr()
    {
        return this->ddr;
    }


    ~Ram()
    {


    }
};

File: python/Ram.py
from Komponen import Komponen

class Ram(Komponen):
    #atribut private
    __kapasitas_gb = ""
    __ddr = ""

    #konstruktor
    def __init__(self, kapasitas_gb, ddr, merk, nama):
        super().__init__(merk, nama)
        self._kapasitas_gb = kapasitas_gb
        self._ddr = ddr

    #setter
    def set_kapasitas_gb(self, kapasitas_gb):
        self._kapasitas_gb = kapasitas_gb

    def set_ddr(self, ddr):
        self._ddr = ddr

    #getter
    def get_kapasitas_gb(self):
        return self._kapasitas_gb

    def get_ddr(self):
        return self._ddr